File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int checksum(long card);
long placeholder = 10;
long placeholder2 = 1;
int count = 0;
int sum1 = 0;
int sum2 = 0;
int totalSum = 0;
long num1 = 0;
long num2 = 0;

int main(void)
{
    long cardNum = get_long("Card: ");

    int sum = checksum(cardNum);

	long digitCount = cardNum;

	for (long digits = 0; digitCount > 0; digits++){
	digitCount /= 10;
	count++;
	}

    if((sum % 10) == 0){
        if((count) == 13){
			printf("VISA\n");
		}else if(count == 15 && ((cardNum / 10000000000000 == 34) || (cardNum / 10000000000000 == 37))){
			printf("AMEX\n");
		}else if(count == 16){
			if (cardNum / 1000000000000000 < 5){
				printf("VISA\n");
			}else if((cardNum / 100000000000000 < 56) && (cardNum / 100000000000000 > 50)){
				printf("MASTERCARD\n");
			}else{
				printf("INVALID\n");
			}
		}else{
			printf("INVALID\n");
		}
    }else{
        printf("INVALID\n");
    }
}

int checksum(long card){

long cardnum = card / 10;
long cardnum2 = card;

for (int i = 0; cardnum > 0; i++){
	long remainder = cardnum % 10; //gets tens place remainder
	cardnum /= 100; //shifts card value by two places

	//for every iteration, need a multiplier to put remainder in correct space

	num1 += remainder * placeholder;

	placeholder *= 100;
}

for (int k = 0; cardnum2 > 0; k++){ // GIVES EVERY OTHER NUMBER STARTING FROM SECOND TO LAST
	long remainder2 = cardnum2 % 10; //gets tens place remainder
	cardnum2 /= 100; //shifts card by two places

	//for every iteration, need a multiplier to put remainder in correct space

	sum2 += remainder2;

	num2 += remainder2 * placeholder2;

	placeholder2 *= 100;
}
// stores numbers: repeat the loop but make each remainder multiply by two and store into an array

int remainders[30];

for (int j = 1; num1 > 0; j++) { //loop with array
	remainders[j] = (num1 % 10) * 2;
	num1 /= 10;
	if (remainders[j] >= 10){
		int x = remainders[j];
		// printf("Iteration: %li\n", j);
		// printf("Value: %li\n", x);
		sum1 += x % 10;
		x /= 10;
		sum1 += x % 10;
	}else{
		sum1 += remainders[j];
	}
}
// GIVES SUM OF MULTIPLIED DIGITS

totalSum = sum1 + sum2;

// printf("Sum 1: %li\n", sum1);
// printf("Sum 2: %li\n", sum2);
//printf("Total Sum: %i\n", totalSum);

return totalSum;
}

File: sentimental-credit/credit.py
from cs50 import get_string

def main():

     strCard = get_string("Card Number: ").strip()

     one = strCard[0]

     two = strCard[0] + strCard[1]

     sOne = int(one)

     sTwo = int(two)

     valid = checksum(strCard)

     digits = len(strCard)

     brand = ""

     if(valid):
          if digits == 13:
               brand = "VISA"
          elif digits == 15 and (two == "34" or two =="37"):
               brand = "AMEX"
          elif digits == 16:
               if sOne < 5:
                    brand = "VISA"
               elif (sTwo < 56) and (sTwo > 50):
                    brand = "MASTERCARD"
               else:
                    brand = "INVALID"
          else:
               brand = "INVALID"
     else:
          brand = "INVALID"

     print(brand)

def checksum(strCard):
     card = int(strCard)
     holder = 10
     holder2 = 1
     sum1 = 0
     sum2 = 0
     num1 = 0
     num2 = 0
     card1 = card // 10
     card2 = card

     while card1 > 0:
          remainder = card1 % 10
          card1 //= 100

          num1 += remainder * holder

          holder *= 100
     while card2 > 0:
          remainder2 = card2 % 10
          card2 //= 100

          sum2 += remainder2

          num2 += remainder2 * holder2

          holder2 *= 100

     remainders = []

     i = 0

     while num1 > 0:
          remainders.append((num1 % 10) * 2)
          num1 //= 10

          if remainders[i] >= 10 :
               x = remainders[i]

               sum1 += x % 10

               x //= 10

               sum1 += x % 10
          else:
               sum1 += remainders[i]

          i += 1

     if (sum1 + sum2) % 10 == 0:
          valid = True
     else:
          valid = False

     return valid

if __name__ == '__main__':
     main()